src/contacts/processors/impl/SearchActionProcessor.java

```diff
@@ -0,0 +1,80 @@
+package contacts.processors.impl;
+
+import contacts.ConsoleReader;
+import contacts.Constants;
+import contacts.model.Organization;
+import contacts.model.Person;
+import contacts.model.Record;
+import contacts.processors.IActionProcessor;
+import contacts.processors.IProcessorFactory;
+import contacts.service.IRecordService;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class SearchActionProcessor implements IActionProcessor {
+
+    private final IRecordService recordService;
+    private final IProcessorFactory processorFactory;
+
+    public SearchActionProcessor(IRecordService recordService, IProcessorFactory processorFactory) {
+        this.recordService = recordService;
+        this.processorFactory = processorFactory;
+    }
+
+    @Override
+    public boolean doAction() throws IOException {
+        String query = ConsoleReader.getStringFromConsole("Enter search query: ");
+        List<Record> resultsList = searchMatching(query);
+
+        int index = 1;
+        System.out.println("Found " + resultsList.size() + " results:");
+
+        for (Record record : resultsList) {
+            if (record instanceof Person person) {
+                System.out.println(index + ". " + person.getName() + Constants.DELIMETER + person.getSurname());
+                index += 1;
+            } else if (record instanceof Organization organization) {
+                System.out.println(index + ". " + organization.getName());
+                index += 1;
+            }
+        }
```

Comment:
I thought we discussed that this approach is not acceptable

Reply:
Done